src/pdo/sink.rs

```diff
@@ -143,25 +152,33 @@ pub struct FixedData {
     pub operational_current_ma: u16,
 }
 
-impl TryFrom<u32> for FixedData {
-    type Error = PdError;
-
-    fn try_from(value: u32) -> Result<Self, Self::Error> {
-        if PdoKind::from(value) != PdoKind::Fixed {
-            return Err(PdError::InvalidParams);
-        }
-
-        let raw = FixedRaw(value);
-        Ok(FixedData {
+impl From<FixedRaw> for FixedData {
+    fn from(raw: FixedRaw) -> Self {
+        FixedData {
             dual_role_power: raw.dual_role_power() != 0,
             higher_capability: raw.higher_capability() != 0,
             unconstrained_power: raw.unconstrained_power() != 0,
             usb_comms_capable: raw.usb_comms_capable() != 0,
             dual_role_data: raw.dual_role_data() != 0,
-            frs_required_current: raw.frs_required_current().into(),
+            frs_required_current: FrsRequiredCurrent::from(raw.frs_required_current()),
             voltage_mv: raw.voltage() * MV50_UNIT,
             operational_current_ma: raw.operational_current() * MA10_UNIT,
-        })
+        }
+    }
+}
+
+impl TryFrom<u32> for FixedData {
+    type Error = ExpectedPdo;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        if PdoKind::from(value) == PdoKind::Fixed {
+            Ok(FixedRaw(value).into())
+        } else {
+            Err(ExpectedPdo {
+                kind: PdoKind::Fixed,
```

Comment:
We expected a `Fixed` but got something else; what did we get? That is useful information in the error for debugging. Can we include it?